pkg/block/index.go

```diff
@@ -358,38 +459,32 @@ func sanitizeChunkSequence(chks []chunks.Meta, mint int64, maxt int64) ([]chunks
 		return chks[i].MinTime < chks[j].MinTime
 	})
 
-	// Remove duplicates and complete outsiders.
+	// Remove duplicates, complete outsiders and near outsiders.
 	repl := make([]chunks.Meta, 0, len(chks))
-	for i, c := range chks {
-		if c.MinTime > maxt || c.MaxTime < mint {
-			// "Complete" outsider. Ignore.
-			continue
-		}
+	var last *chunks.Meta
+	for _, c := range chks {
+		var (
+			err    error
+			ignore = false
+		)
+
+		for _, ignoreChkFn := range ignoreChkFns {
+			ignore, err = ignoreChkFn(mint, maxt, last, &c)
```

Comment:
some ignore check functions will fail if last is nil, is this ok?

Reply:
yes - it is all handled.